sympy/geometry/polygon.py

```diff
@@ -164,42 +169,53 @@ def __new__(cls, *args, **kwargs):
         else:
             return Point(*vertices, **kwargs)
 
-        # reject polygons that have intersecting sides unless the
-        # intersection is a shared point or a generalized intersection.
-        # A self-intersecting polygon is easier to detect than a
-        # random set of segments since only those sides that are not
-        # part of the convex hull can possibly intersect with other
-        # sides of the polygon...but for now we use the n**2 algorithm
-        # and check if any side intersects with any preceding side,
-        # excluding the ones it is connected to
         try:
```

Comment:
convex is no longer used. But if intersecting polygons are allowed, that method needs to be modified to check that there are no collinear segments (which should only happen if a point was needed for a self intersecting polygon). It might be better to store this attribute (is_self_intersecting) at instantiation rather than have to compute it again for testing whether the polygon is convex or not. For that matter, should the convex attribute be stored at instantiation?